### PR TITLE
Allow using existing openstack keys for CI

### PR DESCRIPTION
**Pull request by Itxaka**
This allows to pass a key_pair to the terraform vars for the
openstack deployment which uses an existing key on openstack to
inject into the nodes

## Why is this PR needed?

Does it fix an issue? addresses a business case?

add a description and link to the issue if one exists.

Fixes #

**Reminder**: Add the "fixes bsc#XXXX" to the title of the commit so that it will
appear in the changelog.

## What does this PR do?

Allows to re-use an existing key on the openstack infrastructure to inject into the machines

## Anything else a reviewer needs to know?

This does not change existing behaviour

## Info for QA

This is info for QA so that they can validate this. This is **mandatory** if this PR fixes a bug.
If this is a new feature, a good description in "What does this PR do" may be enough.

### Related info

Info that can be relevant for QA:
* link to other PRs that should be merged together
* link to packages that should be released together
* upstream issues

### Status **BEFORE** applying the patch

How can we reproduce the issue? How can we see this issue? Please provide the steps and the prove
this issue is not fixed.

### Status **AFTER** applying the patch

How can we validate this issue is fixed? Please provide the steps and the prove this issue is fixed.

## Docs

If docs need to be updated, please add a link to a PR to https://github.com/SUSE/doc-caasp.
At the time of creating the issue, this PR can be work in progress (set its title to [WIP]),
but the documentation needs to be finalized before the PR can be merged. 

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

File: ci/infra/openstack/variables.tf [hunk 1]
```diff
@@ -85,6 +85,11 @@ variable "authorized_keys" {
   description = "SSH keys to inject into all the nodes"
 }
 
+variable "key_pair" {
+  default     = ""
+  description = "SSH key stored in openstack to created the nodes with"
```
Comment (ereslibre) [inline ci/infra/openstack/variables.tf]:
```suggestion
  description = "SSH key stored in openstack to create the nodes with"
```

**Comment by Itxaka** [comment]
> Aside from the typo, LGTM, thanks @Itxaka!
@ereslibre integration-tests step is failing, Im not sure why as it claims that:

`make: *** No rule to make target 'test_integration'.  Stop.`

Im not sure this PR has anything to do with that? I hope!

File: ci/infra/openstack/variables.tf (hunk 1 above)
Comment (Itxaka) [reply ci/infra/openstack/variables.tf]: fixed!

**Comment by ereslibre** [comment]
> make: *** No rule to make target 'test_integration'. Stop.

No, can you please rebase on top of master?

**Comment by Itxaka** [comment]
> No, can you please rebase on top of master?

done!

**Comment by pablochacin** [comment]
LGTM. Should we create another PR to expose this parameter to the testurnner's configuration? It should be pretty simple change, but needs some minor refactoring to put it nicely in the code. I can take look at it.

**Comment by Itxaka** [comment]
Is it possible to merge this now that caasp4 is released? :D

**Pull request merged**